js/src/name-pubsub/cancel.js

```diff
@@ -63,24 +52,30 @@ module.exports = (createCommon, options) => {
 
     it('should cancel a subscription correctly returning true', function (done) {
       this.timeout(300 * 1000)
-      const ipnsPath = `/ipns/${nodeId}`
-
-      series([
-        (cb) => ipfs.name.pubsub.subs(cb),
-        (cb) => ipfs.name.publish(value, { resolve: false }, cb),
-        (cb) => ipfs.name.resolve(nodeId, cb),
-        (cb) => ipfs.name.pubsub.subs(cb),
-        (cb) => ipfs.name.pubsub.cancel(ipnsPath, cb),
-        (cb) => ipfs.name.pubsub.subs(cb)
-      ], (err, res) => {
+      const id = 'QmNP1ASen5ZREtiJTtVD3jhMKhoPb1zppET1tgpjHx2NGA'
+      const ipnsPath = `/ipns/${id}`
+
+      ipfs.name.pubsub.subs((err, res) => {
         expect(err).to.not.exist()
-        expect(res).to.exist()
-        expect(res[0]).to.eql([]) // initally empty
-        expect(res[4]).to.have.property('canceled')
-        expect(res[4].canceled).to.eql(true)
-        expect(res[5]).to.be.an('array').that.does.not.include(ipnsPath)
+        expect(res).to.eql([]) // initally empty
```

Comment:
Since the node is shared between tests we can't assume this will be the case. We should assert that the thing you don't want to be in there is not there instead.